### PR TITLE
WIP: Trigger gpexpand after ICW.

### PR DESCRIPTION
**Pull request by pivotal-ning-yu**
The pipeline job icw_planner_centos7_online_expand tests gpexpand in
such a way, it creates a cluster with only 2 primary+mirror segment
pairs, then expand it to 3 and run ICW to verify the expanded cluster
works correctly.

Now we further expand the cluster size from 3 to 4 after ICW, to verify
that tables can be successfully redistributed.

Also renamed the job to icw_planner_centos7_expand.

## Here are some reminders before you submit the pull request
- ~[ ] Add tests for the change~
- ~[ ] Document changes~
- ~[ ] Communicate in the mailing list if needed~
- [ ] Pass `make installcheck`

### CODE REVIEW

**Comment by ashwinstar** [comment]
PR description is only explaining what and missing to explain why?

**Comment by pivotal-ning-yu** [comment]
On Thu, Dec 27, 2018 at 3:37 PM Ashwin Agrawal <notifications@github.com>
wrote:

> PR description is only explaining what and missing to explain why?
>
Thanks for the reminder, updated the commit message and PR description.

File: concourse/scripts/common.bash [hunk 1]
```diff
@@ -54,19 +82,13 @@ function make_cluster() {
...
-  then
+  if [ "$TEST_EXPAND" = "true" ]; then
     su gpadmin -c "make create-demo-cluster NUM_PRIMARY_MIRROR_PAIRS=2"
-    # Backup master pid, to check it after the whole test, be sure there is
-    # no test case restarting the cluster
-    su gpadmin -c "head -n 1 $MASTER_DATA_DIRECTORY/postmaster.pid > master.pid.bk"
-    gen_gpexpand_input
-    su gpadmin -c "createdb gpstatus"
-    su gpadmin -c "gpexpand -i input -D gpstatus"
+    expand_cluster 2 3
```
[the remark this answers is not in the document]
Comment (pivotal-ning-yu) [reply concourse/scripts/common.bash]: Nice point.  Will update the PR.

[the remark this answers is not in the document]
Comment (pivotal-ning-yu) [reply concourse/scripts/common.bash]: Updated the commit, refactored the structure a bit, moved `expand_cluster` outside.

**Comment by pivotal-ning-yu** [comment]
We have a devel pipeline for this PR, it will trigger a failure like below (timestamps are stripped):

```diff
[INFO]:-Unlocked catalog
[INFO]:-Creating expansion schema
[ERROR]:-gpexpand failed: error 'ERROR:  could not access status of transaction 4096 (slru.c:896)  (seg3 10.254.0.150:25438 pid=99714) (slru.c:896)
DETAIL:  Could not read from file "pg_distributedlog/0000" at offset 32768: Success.
' in 'CREATE SCHEMA gpexpand' 

Exiting...
[ERROR]:-gpexpand is past the point of rollback. Any remaining issues must be addressed outside of gpexpand.
[INFO]:-Shutting down gpexpand...
```

Here is a 100% reproducer:

> 1. create a demo cluster
> 2. run regress/autovacuum_template0;
> 3. run expand --(throw this error)

There was a similar issue on master branch which was fixed in https://github.com/greenplum-db/gpdb/pull/6361 , we ever rebased our PR to included that fix, but the issue still can be triggered.

We have a WIP fix/workaround for it, I'll cherry-pick it to this PR to verify the effect.

**Comment by pivotal-ning-yu** [comment]
Pushed that WIP commit.  There is no much commit message and comment in that commit, we'll add them later.

**Comment by pivotal-ning-yu** [comment]
Updated the commits:
- updated the WIP commit for the distributedlog issue, it's cherry-picked from https://github.com/greenplum-db/gpdb/pull/6631;
- rebased with latest feature branch;
- split the pipeline job to two, one using the ctas method, the other using the reshuffle method;
- check the result after expansion;

**Comment by pivotal-ning-yu** [comment]
As the reshuffle devel pipeline is already green we will merge it on the feature branch.  We have separate tracker story to track the remaining issues on ctas.  Thanks for the review.

**Pull request closed without merging**